src/system.jl

```diff
@@ -0,0 +1,302 @@
+"""
+    ServicesTimeSeries(reg, spin, sup_on, sup_off)
+
+Type defining services time series.  Fields are KeyedArray where the keys are generator names
+x datetimes.  Only generators that provide each service are included in the array.
+"""
+struct ServicesTimeSeries
+    reg::KeyedArray{Float64}
+    spin::KeyedArray{Float64}
+    sup_on::KeyedArray{Float64}
+    sup_off::KeyedArray{Float64}
+end
+
+"""
+    Zone(number, reg, spin, sup_on, sup_off)
+
+Type defining a market zone.  The `Zone` is identified by a number.  The other fields contain
+the service requirements for the zone.
+"""
+struct Zone
+    number::Int64
+    reg::Float64
+    spin::Float64
+    sup_on::Float64
+    sup_off::Float64
+end
+
+###### Static Component Types ######
+
+abstract type StaticComponent end
+
+Base.length(components::StaticComponent) = length(getfield(components, 1))
+# define interfaces? Iterator? Table?
+
+"""
+    Generators(
+        name::Vector{Int}
+        zone::Vector{Int}
+        startup_cost::Vector{Float64}
+        shutdown_cost::Vector{Float64}
+        no_load_cost::Vector{Float64}
+        time_at_status::Vector{Float64}
+        min_uptime::Vector{Float64}
+        min_downtime::Vector{Float64}
+        ramp_up::Vector{Float64}
+        ramp_down::Vector{Float64}
+        initial_gen::Vector{Float64}
+        technology::Vector{Symbol}
+    )
+
+Type for static generator component attributes (i.e. things that describe a generator that
+are not time series data).
+"""
+struct Generators <: StaticComponent
```

Comment:
How do these types compare to those in PowerFlowData.jl? Does this have any info not contained in `PowerFlowData.Generators`?

Is it viable to have a constructor `Generators(::PowerFlowData.Generators) = ...`?